File: WS_JAVA_REST_BanQuito/src/main/java/ec/edu/monster/model/Movimiento.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.model;

/**
 *
 * @author danie
 */

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "movimiento")
public class Movimiento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "num_cuenta")
    private Cuenta cuenta;

    @Column(length = 3, nullable = false)
    private String tipo; // DEP, RET, TRA

    @Column(length = 10, nullable = false)
    private String naturaleza; // INGRESO, EGRESO

    @Column(name = "interno_transferencia", nullable = false)
    private boolean internoTransferencia = false;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal valor;

    @Column(nullable = false)
    private LocalDate fecha;

    // Getters y setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Cuenta getCuenta() {
        return cuenta;
    }

    public void setCuenta(Cuenta cuenta) {
        this.cuenta = cuenta;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getNaturaleza() {
        return naturaleza;
    }

    public void setNaturaleza(String naturaleza) {
        this.naturaleza = naturaleza;
    }

    public boolean isInternoTransferencia() {
        return internoTransferencia;
    }

    public void setInternoTransferencia(boolean internoTransferencia) {
        this.internoTransferencia = internoTransferencia;
    }

    public BigDecimal getValor() {
        return valor;
    }

    public void setValor(BigDecimal valor) {
        this.valor = valor;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }
}

File: PROYECTO/01.SERVIDOR/WS_DOTNET_SOAP_BanQuito/WS_DOTNET_SOAP_BanQuito/ec.edu.monster.model/Movimiento.cs
using System;
using System.Collections.Generic;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WS_DOTNET_SOAP_BanQuito.ec.edu.monster.model
{
    [Table("movimiento")]
    public class Movimiento
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("num_cuenta")]
        [StringLength(20)]
        [ForeignKey("Cuenta")]
        public string NumCuenta { get; set; }

        public virtual Cuenta Cuenta { get; set; }

        [Required]
        [Column("tipo")]
        [StringLength(3)]
        public string Tipo { get; set; } // DEP, RET, TRA

        [Required]
        [Column("naturaleza")]
        [StringLength(10)]
        public string Naturaleza { get; set; } // INGRESO, EGRESO

        [Required]
        [Column("interno_transferencia")]
        public bool InternoTransferencia { get; set; } = false;

        [Required]
        [Column("valor", TypeName = "decimal")]
        public decimal Valor { get; set; }

        [Required]
        [Column("fecha", TypeName = "date")]
        public DateTime Fecha { get; set; }
    }
}
